connectors/src/connectors/google_drive/index.ts

```diff
@@ -304,43 +308,221 @@ export async function retrieveGoogleDriveConnectorPermissions(
+      let remoteFolders: drive_v3.Schema$File[] = [];
+      do {
+        const res: GaxiosResponse<drive_v3.Schema$FileList> =
+          await drive.files.list({
+            corpora: "allDrives",
+            pageSize: 200,
+            includeItemsFromAllDrives: true,
+            supportsAllDrives: true,
+            fields:
+              "nextPageToken, files(id, name, parents, mimeType, createdTime, modifiedTime, trashed, webViewLink)",
+            q: `'${parentInternalId}' in parents and mimeType='application/vnd.google-apps.folder'`,
+            pageToken: nextPageToken,
           });
-          fd.name = d.data.name;
+
+        if (res.status !== 200) {
+          throw new Error(
+            `Error getting files. status_code: ${res.status}. status_text: ${res.statusText}`
+          );
         }
-        return {
-          provider: c.type,
-          internalId: f.folderId,
-          parentInternalId: null,
-          type: "folder" as ConnectorResourceType,
-          title: fd.name || "",
-          sourceUrl: fd.webViewLink || null,
-          expandable: false,
-          permission: "read" as ConnectorPermission,
-        };
-      })();
-    })
-  );
+        if (!res.data.files) {
+          throw new Error("Files list is undefined");
+        }
+        remoteFolders = remoteFolders.concat(res.data.files);
+        nextPageToken = res.data.nextPageToken || undefined;
+      } while (nextPageToken);
+      const resources: ConnectorResource[] = await Promise.all(
+        remoteFolders.map(async (rf): Promise<ConnectorResource> => {
+          const driveObject = await driveObjectToDustType(rf, authCredentials);
+
+          return {
+            provider: c.type,
+            internalId: driveObject.id,
+            parentInternalId: driveObject.parent,
+            type: "folder" as ConnectorResourceType,
+            title: driveObject.name,
+            sourceUrl: driveObject.webViewLink || null,
+            expandable: await folderHasChildren(connectorId, driveObject.id),
+            permission: (await GoogleDriveFolders.findOne({
+              where: {
+                connectorId: connectorId,
+                folderId: driveObject.id,
+              },
+            }))
+              ? "read"
+              : "none",
+          };
+        })
+      );
 
-  return new Ok(resources);
+      return new Ok(resources);
+    }
+  } else {
+    return new Err(new Error(`Invalid permission: ${filterPermission}`));
+  }
+}
+
+export async function setGoogleDriveConnectorPermissions(
+  connectorId: ModelId,
+  permissions: Record<string, ConnectorPermission>
+): Promise<Result<void, Error>> {
+  const connector = await Connector.findByPk(connectorId);
+  if (!connector) {
+    return new Err(new Error(`Connector not found with id ${connectorId}`));
+  }
+
+  let shouldFullSync = false;
+  for (const [id, permission] of Object.entries(permissions)) {
+    shouldFullSync = true;
+    if (permission === "write") {
```

Comment:
We should not support `write` or `read_write` we should error if we get these. It should be `read` or `none`